lua/randomat2/events/farfetched.lua

```diff
@@ -1,45 +1,84 @@
 local EVENT = {}
 
 EVENT.Title = "Farfetched"
-EVENT.Description = "The Magneto-stick has unlimited range and can lift much heavier objects than usual"
+EVENT.Description = "The Magneto-stick has greatly extended range and can lift much heavier objects than usual"
 EVENT.id = "farfetched"
 EVENT.Categories = {"fun", "smallimpact"}
 
-CreateConVar("randomat_farfetched_rangemult", 2.5, FCVAR_NONE, "The multiplier to use on the magneto-stick's range", 1, 10)
-CreateConVar("randomat_farfetched_forcemult", 2.5, FCVAR_NONE, "The multiplier to use on the magneto-stick's lift force", 1, 10)
+CreateConVar("randomat_farfetched_rangemult", 25, FCVAR_NONE, "The multiplier to use on the magneto-stick's range", 1, 50)
+CreateConVar("randomat_farfetched_weightmult", 10, FCVAR_NONE, "The multiplier to use on the magneto-stick's maximum carry weight", 1, 50)
 
-local function SetupWeapon(wep, rangemult)
+local function SetupWeapon(wep, rangemult, weightmult)
     if not IsValid(wep) then return end
-    if wep.OldGetRange then return end
+    if wep.OldGetRange or wep.OldThink then return end
 
     wep.OldGetRange = wep.GetRange
     wep.GetRange = function(this, target)
         local range = this:OldGetRange(target)
         return range * rangemult
     end
+
+    local ent_diff_time = 0
+    local stand_time = 0
+
+    wep.OldThink = wep.OldThink
+    wep.Think = function(this)
+        this.BaseClass.Think(this)
+        if not this:CheckValidity() then return end
+
+        if CurTime() > ent_diff_time then
+            if this:GetPos():DistToSqr(this.EntHolding:GetPos()) > 40000 * rangemult * rangemult then
+                this:Reset()
+                return
+            end
+            ent_diff_time = CurTime() + 1
+        end
+
+        if CurTime() > stand_time then
+            for _, p in player.Iterator() do
+                if p:IsTerror() and p:GetGroundEntity() == this.EntHolding then
+                    this:Reset()
+                    return
+                end
+            end
+            stand_time = CurTime() + 0.1
+        end
+
+        this.CarryHack:SetPos(this:GetOwner():EyePos() + this:GetOwner():GetAimVector() * 70)
+        this.CarryHack:SetAngles(this:GetOwner():GetAngles())
+        this.EntHolding:PhysWake()
+    end
+
+    CARRY_WEIGHT_LIMIT = CARRY_WEIGHT_LIMIT * weightmult
```

Comment:
This will increase the carry limit for every magneto stick that is modified. It should be moved into `Begin` and reverted in `End` instead of `RevertWeapon`